lib/exceptions.h

```diff
@@ -54,27 +60,40 @@ class CompilerBug final : public P4CExceptionBase {
  public:
     template <typename... T>
     CompilerBug(const char* format, T... args)
-            : P4CExceptionBase(format, args...)
-    { message = cstring(ANSI_RED) + "Compiler Bug" + ANSI_CLR + ":\n" + message; }
+            : P4CExceptionBase(format, args...) {
+        // Check if output is redirected and if so, then don't color text so that
+        // escape characters are not present
+        message = (is_cerr_redirected() ? "" : cstring(ANSI_RED)) + "Compiler Bug"
```

Comment:
I suggested moving the entire expression `(is_cerr_redirected() ? "" : cstring(ANSI_RED))` into a helper function. Perhaps colorize(ANSI_RED) and clear_colors().